### PR TITLE
refactor: port ma layout to ts

### PR DESCRIPTION
**Pull request by alex-castelo**
This is how I felt about this PR 😂 

image

### CODE REVIEW

File: src/components/MaLayout/MaLayout.vue [hunk 1]
```diff
@@ -160,14 +209,23 @@ function renderGrid({ createElement, dom, grid, gap, justify, verticalAlign }) {
   return rows
 }
 
+interface RenderColumns {
```
Comment (sergigraciaj) [inline src/components/MaLayout/MaLayout.vue:212]: Should be move this interface before the Vue.Extends, near the types, and rename it to IRenderColumns? Or just use it as a a type.

File: src/components/MaLayout/MaLayout.vue [hunk 2]
```diff
@@ -160,14 +209,23 @@ function renderGrid({ createElement, dom, grid, gap, justify, verticalAlign }) {
...
+  createElement: CreateElement
+  columns: string[] | string
```
Comment (sergigraciaj) [inline src/components/MaLayout/MaLayout.vue:214]: LayoutColumns type here? 🤔

File: src/components/MaLayout/MaLayout.vue (hunk 1 above)
Comment (sergigraciaj) [reply src/components/MaLayout/MaLayout.vue:212]: ^Same for RenderStack, RenderGrid

File: src/components/MaLayout/MaLayout.vue [hunk 3]
```diff
@@ -108,14 +126,27 @@ export default {
...
-}
+})
 
-function getGrid({ parent, columns }) {
+function getGrid({
+  parent,
+  columns,
+}: {
+  parent: Vue
+  columns: string
```
Comment (sergigraciaj) [inline src/components/MaLayout/MaLayout.vue:136]: Columns can be an Array too, so LayoutColumns type here and also on arguments in line 110, right?

File: src/components/MaLayout/MaLayout.vue (hunk 1 above)
Comment (alex-castelo) [reply src/components/MaLayout/MaLayout.vue:212]: We have talked before with Edu that Interface should be something that is extended or implemented. Let's keep the types for components for now (it's all debatable, I want to do some more research on this).

I agree to move everything up near the type declarations (even create a MaLayout type file).

**Review by sergigraciaj — APPROVED** [review]

**Pull request merged**